### PR TITLE
Vs refactor wire up ranges and date ranges

### PR DESCRIPTION
**Pull request by not-napoleon**
Pretty straightforward wiring up.  Since both `RangeAggregatorFactory` and `DateRangeAggregatorFactory` are thin wrappers over the base `AbstractRangeFactory`, I put the registration logic there.  I did have to fix two exception messages in the tests, but didn't have to change any exception types.

### CODE REVIEW

**Comment by not-napoleon** [comment]
@elasticmachine run elasticsearch-ci/1

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/AbstractRangeAggregatorFactory.java [hunk 1]
```diff
@@ -42,20 +46,44 @@
     private final InternalRange.Factory<?, ?> rangeFactory;
     private final R[] ranges;
     private final boolean keyed;
+    private final String aggregationTypeName;
 
+    public static void registerAggregators(ValuesSourceRegistry valuesSourceRegistry, String aggregationName) {
+        valuesSourceRegistry.register(aggregationName,
+            List.of(CoreValuesSourceType.NUMERIC, CoreValuesSourceType.DATE, CoreValuesSourceType.BOOLEAN),
```
Comment (csoulios) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/AbstractRangeAggregatorFactory.java]: Is Boolean range a valid concept?

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/AbstractRangeAggregatorFactory.java [hunk 2]
```diff
@@ -73,13 +101,14 @@ protected Aggregator doCreateInternal(ValuesSource valuesSource,
...
                                             List<PipelineAggregator> pipelineAggregators,
                                             Map<String, Object> metaData) throws IOException {
-        if (valuesSource instanceof Numeric == false) {
-            throw new AggregationExecutionException("ValuesSource type " + valuesSource.toString() + "is not supported for aggregation " +
-            this.name());
+
+        AggregatorSupplier aggregatorSupplier = queryShardContext.getValuesSourceRegistry().getAggregator(config.valueSourceType(),
+            aggregationTypeName);
+        if (aggregatorSupplier instanceof RangeAggregationSupplier == false) {
+            throw new AggregationExecutionException("Registry miss-match - expected HistogramAggregatorSupplier, found [" +
```
Comment (csoulios) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/AbstractRangeAggregatorFactory.java]: Copy-paste nit: Exception message should read RangeAggregationSupplier, not HistogramAggregatorSupplier.

**Review by csoulios — APPROVED** [review]
LGTM. I left couple comments.

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/AbstractRangeAggregatorFactory.java (hunk 1 above)
Comment (not-napoleon) [reply server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/AbstractRangeAggregatorFactory.java]: That's a really good question.  I think master will let you do this, which is why I included it, but I agree it's kind of a nonsense operation. My strategy so far has been to keep these things supported for now, and after we merge have a conversation about what we want to do long term with these cases.

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/AbstractRangeAggregatorFactory.java (hunk 2 above)
Comment (not-napoleon) [reply server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/AbstractRangeAggregatorFactory.java]: Thanks, will fix.

**Pull request merged**